mesonbuild/compilers/mixins/clang.py

```diff
@@ -176,6 +176,9 @@ def get_has_func_attribute_extra_args(self, name: str) -> T.List[str]:
         # error.
         return ['-Werror=attributes']
 
+    def get_prelink_args(self, prelink_name: str, obj_list: T.List[str]) -> T.Tuple[T.List[str], T.List[str]]:
+        return [prelink_name], ['-r', '-o', prelink_name] + obj_list
```

Comment:
If prelinking was broken in some versions of Clang, this most likely needs to be guarded by a version check.

Reply:
Agreed, I will dip up old Clang versions to check when it started working. I will also add the check to the test suite then similar to GCC. (On a semi-related note, maybe for GCC the version should be checked here as well, and not just in the test suite?)